.github/workflows/test-sarif.yaml

```diff
@@ -0,0 +1,61 @@
+name: Test kube-linter SARIF output
+
+on:
+  - push
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Checkout all repo history to make tags available for figuring out kube-linter version during build.
+          fetch-depth: 0
+
+      - name: Read Go version from go.mod
+        run: echo "GO_VERSION=$(grep -E "^go\s+[0-9.]+$" go.mod | cut -d " " -f 2)" >> $GITHUB_ENV
+      - name: Setup Go environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache Go modules
+        uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            ${{ runner.os }}-go-
+
+      - name: Build binaries
+        run: make build
+
+      - name: Print kube-linter version
+        run: .gobin/kube-linter version
+
+      - name: Run kube-linter on a sample file with SARIF output
+        run: .gobin/kube-linter lint --format=sarif e2etests/testdata/inputs/splunk.yaml > results.sarif
+        continue-on-error: true
```

Comment:
We do a bunch of this stuff already in CircleCI. I wonder if we can just move this test over there, into the `e2e-tests`?

Reply:
The key is the last step which runs `github/codeql-action/upload-sarif@v1` action.

I can't find a "mainstream" way to call a github action from CircleCI so that we could move this workflow in CircleCI entirely. In the end, it is probably just some nodejs code and I can look for way to invoke it stand-alone if you think it's worth it.

`github/codeql-action/upload-sarif@v1` calls some REST endpoint and theoretically we could call it from CircleCI. However it also generates `partialFingerprints` for de-duplicating alerts which logic I don't think is worth replicating by hand.

> If you upload a SARIF file using the upload-sarif action and this data is missing, GitHub attempts to populate the partialFingerprints field from the source files.
> If you upload a SARIF file without fingerprint data using the /code-scanning/sarifs API endpoint, the code scanning alerts will be processed and displayed, but users may see duplicate alerts.

(from https://docs.github.com/en/code-security/secure-coding/sarif-support-for-code-scanning#preventing-duplicate-alerts-using-fingerprints)

A third option could be to run all steps in CircleCI and only the last one, SARIF upload, in GitHub Actions. For that GitHub Actions workflow should wait for CircleCI workflow and download `results.sarif` artifact when it is ready. Could be possible. I did not go this way initially because the entire GitHub actions workflow takes only 2m 40s.

What do you think?

Reply:
Let's leave it as is. My initial suggestion was predicated on the removal of the upload-sarif step. We should probably eventually migrate all CI here off CircleCI and into GitHub Actions, because there's a bunch of useful actions we likely want to use.